BackEnd/src/main/java/com/unosquare/admin_core/back_end/repository/EventRepository.java

```diff
@@ -15,9 +18,26 @@
 
     Event findByEmployeeAndStartDateAndEndDateAndEventType(Employee employee, LocalDate startDate, LocalDate endDate, EventType eventType);
 
-    List<Event> findByStartDateBetweenAndEventType(LocalDate rangeStart, LocalDate rangeEnd,EventType eventType);
+    List<Event> findByStartDateBetweenAndEventType(LocalDate rangeStart, LocalDate rangeEnd, EventType eventType);
 
     List<Event> findByEventStatusAndEventType(EventStatus eventStatus, EventType eventType);
 
     List<Event> findByEventType(EventType eventType);
+
+    @Query(value = "SELECT totalHolidays " +
+            "FROM Employee " +
+            "WHERE employeeId = :employeeId")
+    Long getTotalEventsRemainingByEmployee(@Param("employeeId") int employeeId);
+
+    @Query(value = "SELECT COUNT(eventType) " +
+            "FROM Event " +
+            "WHERE " +
+            "employee = :employeeId " +
+            "AND " +
+            "startDate = :startDate " +
+            "AND " +
```

Comment:
I'm not sure if this is the correct approach, surely it needs to intersect? for example if we have 2 ranges 1st - > 3rd and 2nd -> 4th it would not be caught here.